android/src/main/java/com/reactnativecommunity/cookies/CookieManagerModule.java

```diff
@@ -63,36 +78,180 @@ public void getFromResponse(String url, Promise promise) throws URISyntaxExcepti
-                if (cookie.length > 1) {
-                  map.putString(cookie[0].trim(), cookie[1]);
+    public void clearByName(String url, String name, Boolean useWebKit, final Promise promise) {
+        promise.reject(new Exception("Cannot remove a single cookie by name on Android"));
+    }
+
+    @ReactMethod
+    public void clearAll(Boolean useWebKit, final Promise promise) {
+        if (USES_LEGACY_STORE) {
+            mCookieManager.removeAllCookie();
+            mCookieManager.removeSessionCookie();
+            mCookieSyncManager.sync();
+            promise.resolve(true);
+        } else {
+            mCookieManager.removeAllCookies(new ValueCallback<Boolean>() {
+                @Override
+                public void onReceiveValue(Boolean value) {
+                    promise.resolve(value);
                 }
+            });
+            mCookieManager.flush();
+        }
+    }
+
+    private void addCookies(String url, String cookieString, final Promise promise) {
+        try {
+            if (USES_LEGACY_STORE) {
+                mCookieManager.setCookie(url, cookieString);
+                mCookieSyncManager.sync();
+                promise.resolve(true);
+            } else {
+                mCookieManager.setCookie(url, cookieString, new ValueCallback<Boolean>() {
+                    @Override
+                    public void onReceiveValue(Boolean value) {
+                        promise.resolve(value);
+                    }
+                });
             }
+        } catch (Exception e) {
+            promise.reject(e);
         }
-        promise.resolve(map);
     }
 
-    @ReactMethod
-    public void clearAll(boolean useWebKit, final Promise promise) {
-        this.cookieHandler.clearCookies(new Callback() {
-            public void invoke(Object... args) {
-                promise.resolve(null);
+    private WritableMap getCookies(String url) throws Exception {
+
+        Map<String, Cookie> cookieObjects = getCookieObjects(url);
+
+        WritableMap allCookiesMap = Arguments.createMap();
+
+        Set<String> keys = cookieObjects.keySet();
+        for (String key : keys) {
+            Cookie cookie = cookieObjects.get(key);
+            if (cookie != null) {
+                WritableMap cookieMap = Arguments.createMap();
+                cookieMap.putString("name", cookie.name());
+                cookieMap.putString("value", cookie.value());
+                cookieMap.putString("domain", cookie.domain());
+                cookieMap.putString("path", cookie.path());
+                // no version unavailable from this interface
+                // cookieMap.putInt("version", cookie.getVersion());
+                cookieMap.putBoolean("secure", cookie.secure());
+                cookieMap.putBoolean("httpOnly", cookie.httpOnly());
+
+                long expires = cookie.expiresAt();
+                cookieMap.putString("expiration", new Date(expires).toString());
```

Comment:
expiration date should be rfc1123 compliant.
Since `toGMTString()` is deprecated, you should format with this pattern : `EEE, dd MMM yyyy HH:mm:ss zzz`
use `DateTimeFormatter.RFC_1123_DATE_TIME.format(...)` if Java 8 is the min supported java in this project. 
Or use SimpleDateFormatter otherwise :
```java
DateFormat df = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", java.util.Locale.US);
df.setTimeZone(TimeZone.getTimeZone("GMT"));
cookieMap.putString("expiration", df.format(expires));
```

Reply:
Will get that amended and testes as soon as I can.

Reply:
I have addressed this in the format of the cookie string for setting. However for the cookie object passed in for set and the cookie objects returned from get I have matched the iOS formatting `yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ`

Reply:
Since we are making breaking changes, there is an argument to using the same format across the board and complying with the cookie string date format i.e. RFC 1123. 
What do you guys think @Aure77 and @safaiyeh ?

Reply:
Yeah let's keep it consistent across the board

Reply:
Cool, will be able to dedicate some time tomorrow and will swap them all out to use the RFC 1123 format. Currently we are not rejecting cookies if the date fails, which is probably not quite right. I will add that in too. Shall shout out when that is sorted :)